microservices/biobuddy/biobuddy/job_state_manager.py

```diff
@@ -0,0 +1,27 @@
+class JobState:
+    def __init__(self):
+        self.is_running = False
+
+
+class JobStateManager:
+    def __init__(self):
+        self.jobs = {}  # Maps job_id to JobState
+
+    def start_job(self, job_id: str):
+        if job_id not in self.jobs:
+            self.jobs[job_id] = JobState()
+        self.jobs[job_id].is_running = True
+
+    def finish_job(self, job_id: str):
+        if job_id in self.jobs:
+            del self.jobs[job_id]
+
+    def is_job_running(self, job_id: str) -> bool:
+        return self.jobs.get(job_id, JobState()).is_running
```

Comment:
I would rather throw not found there... but we will address this later...

Reply:
Refactoring is like a proctology exam - cold and not that pleasant initially, a bit fun and warm by the end. And, of course, is needed for your health 😁